Frends.Community.LDAP.Tests/UnitTests.cs

```diff
@@ -1,269 +1,291 @@
-        ///  Test for AD_SetUserPassword: Set a user's password.
-        /// </summary>
-        [Test, Order(12)]
-        [Ignore("Test is not working on build server. TODO")]
-
-        public void ShouldSetPassword()
-        {
-            var result = LdapActiveDirectoryOperations.AD_SetUserPassword(_passConnection);
-            Assert.AreEqual(result.OperationSuccessful, true);
-        }
-    }
-}
+            var u = LdapActiveDirectoryOperations.AD_FetchObjects(_connection, e);
+            var result = u[0].GetAccountExpiresDateTime();
+            Assert.AreEqual(expectedDateTime, result);
+        }
+
+        [Test, Order(6)]
+        public void ShouldGetUserLastLogonDateTime()
+        {
+            var e = new AD_FetchObjectProperties()
+            {
+                Filter = "(&(objectClass=user)(cn=" + _user + "))",
+                Path = _path
+            };
+
+            var expectedDateTime = DateTime.MinValue;
+
+            var u = LdapActiveDirectoryOperations.AD_FetchObjects(_connection, e);
+            var result = u[0].GetPropertyDateTime("lastLogon");
+            Assert.AreEqual(expectedDateTime.GetType(), result.GetType());
+        }
+
+        [Test, Order(7)]
+        public void ShouldUpdateUser()
+        {
+            var user = new UpdateADuser { DN = _dn };
+            var attributes = new List<EntryAttribute>
+            {
+                new EntryAttribute {Attribute = AdUserAttribute.description, Value = "MattiMeikalainen", DataType = AttributeType.String}
+            };
+            user.OtherAttributes = attributes.ToArray();
+            var result = LdapActiveDirectoryOperations.AD_UpdateUser(_connection, user);
+            Assert.AreEqual(result.OperationSuccessful, true);
+        }
+
+        [Test, Order(8)]
+        public void ShouldAddGroups()
+        {
+            var u = new AD_AddGroupsUserProperties { Dn = _dn };
+            var e = new AD_AddGroupsProperties { Groups = new[] { _groupDn } };
+
+            var result = LdapActiveDirectoryOperations.AD_AddGroups(_connection, u, e);
+            Assert.AreEqual(result.OperationSuccessful, true);
+        }
+
+        /// <summary>
+        /// Test for AD_SearchOjects: fetch a property and a not loaded property.
+        /// </summary>
+        [Test, Order(8)]
+        public void ShouldSearchUser()
+        {
+            var prop = new AD_SearchObjectProperties()
+            {
+                Filter = "(&(objectClass=user)(cn=" + _user + "))",
+                Path = _path,
+                PropertiesToLoad = new[] { "cn", "invalidProperty" }
+            };
+
+            var ret = LdapActiveDirectoryOperations.AD_SearchObjects(_connection, prop);
+
+            var cnValue = ret[0].GetPropertyStringValue("cn");
+
+            // Should return null.
+            var nullValue = ret[0].GetPropertyStringValue("name");
+            Assert.AreEqual(cnValue, _user);
+            Assert.AreEqual(nullValue, null);

```

Comment:
Please flip the parameters around, since the first parameter should be the expected value instead of real value.